packages/react-client/src/hooks/devices/useInitializeDevices.ts

```diff
@@ -17,13 +18,16 @@ export type InitializeDevicesErrors = { audio: DeviceError | null; video: Device
  * @category Devices
  */
 export const useInitializeDevices = () => {
-  const { videoDeviceManagerRef, audioDeviceManagerRef, hasDevicesBeenInitializedRef } = useFishjamContext();
+  const { videoDeviceManagerRef, audioDeviceManagerRef, devicesInitializationRef } = useFishjamContext();
 
   const initializeDevices: (params?: UseInitializeDevicesParams) => Promise<null | InitializeDevicesErrors> =
     useCallback(
       async ({ enableVideo = true, enableAudio = true }: UseInitializeDevicesParams = {}) => {
-        if (hasDevicesBeenInitializedRef.current) return null;
-        hasDevicesBeenInitializedRef.current = true;
+        if (devicesInitializationRef.current) return null;
+
+        const deferred = new Deferred<void>();
+
+        devicesInitializationRef.current = deferred.promise;
```

Comment:
Should logic in useCallback be wrapped in try/catch. So `deferred.promise` could be rejected if something will fall?